src/github.com/getlantern/flashlight/analytics/analytics.go

```diff
@@ -34,47 +33,61 @@ var (
 
 	maxWaitForIP = math.MaxInt32 * time.Second
 
+	// We get the user agent to use from live data on the proxy, but don't wait
+	// for it forever!
+	maxWaitForUserAgent = 30 * time.Second
+
 	// This allows us to report a real user agent from clients we see on the
 	// proxy.
 	userAgent = eventual.NewValue()
+
+	// Hash of the executable
+	hash = getExecutableHash()
 )
 
 // Start starts the GA session with the given data.
-func Start(cfg *config.Config, version string) func() {
+func Start(deviceID, version string) func() {
+	return start(deviceID, version, geolookup.GetIP, maxWaitForUserAgent, trackSession)
+}
+
+// start starts the GA session with the given data.
+func start(deviceID, version string, ipFunc func(time.Duration) string, uaWait time.Duration,
+	transport func(string, eventual.Getter)) func() {
 	var addr atomic.Value
 	go func() {
 		logging.AddUserAgentListener(func(agent string) {
 			userAgent.Set(agent)
 		})
-		ip := geolookup.GetIP(maxWaitForIP)
+		ip := ipFunc(maxWaitForIP)
 		if ip == "" {
-			log.Errorf("No IP found within %v, not starting analytics session", maxWaitForIP)
-			return
+			log.Errorf("No IP found within %v", maxWaitForIP)
```

Comment:
We used to not report any analytics if we couldn't resolve the IP, which doesn't really make sense because there's lots of other useful info to report. So now we just keep going and don't report the IP if we don't know it.